javascript/react/introduction-to-hooks/rules-of-hooks.md

```diff
@@ -0,0 +1,104 @@
+---
+author: kapnobatai136
+
+levels:
+
+  - medium
+  
+aspects:
+
+  - new
+
+  - workout
+
+  - deep
+
+type: normal
+
+category: how to
+
+links:
+  - '[rules-of-hooks](https://reactjs.org/docs/hooks-rules.html){documentation}'
+
+---
+
+# Rules of hooks
+
+---
+## Content
+
+You have learned how to use the `useState()` and `useEffect()` hooks, but until now we have only called these methods once. An important feature of hooks is that you can have multiple `useState()` and `useEffect()` in the same functional component. Here's a simple example:
+
+```jsx
+function App() {
+  const [foo, setFoo] = useState("First");
+
+  useEffect(() => {
+    localStorage.setItem(
+      "First state",
+      foo
+    );
+  });
+
+  const [bar, setBar] = useState("Second");
+
+  useEffect(function saveData() {
+    localStorage.setItem(
+      "Second state",
+      bar
+    );
+  });
+
+  // ...
+}
+```
+
+Now that you know your functional components can contain multiple hook calls, it is time to learn the rules that apply to hooks.
+
+The first rule is **always call hooks at the top level**. To allow React to correctly preserve the state of hooks between multiple `useState()` and `useEffect` calls, you can't call hooks inside loops, conditions, or nested functions. What this rule does is it ensures that the hooks are called in the exact same order each time a component renders.
```

Comment:
Haven't verified this exact insight but my gut tells me that adding some whitespace would help the readability (feel free to reject this though)

```suggestion
The first rule is **always call hooks at the top level**. 

To allow React to correctly preserve the state of hooks between multiple `useState()` and `useEffect` calls, you can't call hooks inside loops, conditions, or nested functions. What this rule does is it ensures that the hooks are called in the exact same order each time a component renders.
```